### PR TITLE
add: AgentSelect when hideAgent configuration is disabled

### LINKED ISSUE #2589: Provide selecting agent when creating a session in Neo Session launcher.
### Main idea

Since Client-py supports agent-specific session creation, WebUI need to provide this feature as an option when [`hideAgents`](https://github.com/lablup/backend.ai-webui/blob/c4a3f815bef05624687ec6d090cd85913541132b/config.toml.sample#L22) option is disabled.

### Alternative ideas

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by lizable**
Resolves #2589 

**Changes:**
Adds agent selection capability to the session launcher, allowing users to manually choose specific compute agents instead of relying solely on automatic scheduler allocation.

Key additions:
- New `AgentSelect` component displaying available agents with their remaining resource capacities
- Integration into resource allocation form with auto-select option as default
- Agent selection is restricted to single-node single-container environments
- Added translations for agent selection UI elements across multiple languages

When a specific agent is selected:
- Cluster mode is forced to single-node
- Cluster size is set to 1
- The selection is passed to the backend via the `agent_list` parameter

The agent selector includes a refresh button to update the available agent list and displays real-time resource availability for each agent.

**Minimum requirements to check:**
- Agent selection dropdown shows available agents with correct resource information
- Selecting a specific agent properly restricts cluster configuration
- Auto-select option works as default
- Refresh button updates the agent list
- Agent selection is only available when not hidden by configuration

**Test cases:**
> ⚠️ Prerequisites:
> 1. Set the value of  `hideAgents` in config file to false.
> 2. Change the branch of Backend.AI Core to this PR (https://app.graphite.dev/github/pr/lablup/backend.ai/2614/refactor-apply-session-allocation-at-specific-agent-when-hide-agent-option-is-disabled)
> 3. Set the value of `hide-agents` to false in manager.toml file
> 4. Restart Manager process. 

1. Verify agent list displays with resource information
2. Confirm cluster settings are restricted when specific agent is selected
3. Test refresh functionality updates available agents
4. Validate auto-select behavior works as expected

### Screenshot(s)
| After | Before |
|------|--------|
|  |  |

### CODE REVIEW

**Comment by lizable** [comment]
* **#2599** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2599?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lizable and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by yomybaby — CHANGES_REQUESTED** [review]
🙏  Please resolve conflicts. @lizable

File: react/src/components/AgentSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+              );
+            })}
+          </Flex>
+        </Flex>
+      ),
+      value: agent?.id,
+    };
+  });
+
+  console.log(agentOptions);
```
Comment (agatha197) [inline react/src/components/AgentSelect.tsx]:
```suggestion
```

**Review by agatha197 — CHANGES_REQUESTED** [review]
Please resolve the conflicts :)

File: react/src/components/AgentSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import { useTranslation } from 'react-i18next';
+import { useLazyLoadQuery } from 'react-relay';
+
+interface Props extends SelectProps {
+  autoSelectDefault?: boolean;
+  fetchKey?: string;
+  resourceGroup?: string | null;
+}
+
+const AgentSelector: React.FC<Props> = ({
```
Comment (yomybaby) [inline react/src/components/AgentSelect.tsx]: Please rename to "AgentSelect" 😀

**Review by yomybaby — CHANGES_REQUESTED** [review]
> "Currently, agent selection is only available in a single-node single-container environment." 

~~It's better to disable and change the form values related to cluster setting when user set the specific agent.~~Oh, I see you've already handled this. I'll check again when the multi-node test server is ready.

**Review by yomybaby — CHANGES_REQUESTED** [review]
Additional work to be necessary:
- Add agent info to preview step

- Please add a version condition to ensure version compatibility.

Below is not related to this PR's changes, but need to be fix
- When pending status, "null(undefined)" in Agent column

**Review by yomybaby — APPROVED** [review]
LGTM 💪

File: react/src/components/ResourceAllocationFormItems.tsx [hunk 3]
```diff
@@ -16,6 +16,7 @@ import {
   ResourcePreset,
   useResourceLimitAndRemaining,
 } from '../hooks/useResourceLimitAndRemaining';
+import AgentSelector from './AgentSelect';
```
Comment (agatha197) [inline react/src/components/ResourceAllocationFormItems.tsx]:
```suggestion
import AgentSelect from './AgentSelect';
```

File: react/src/components/ResourceAllocationFormItems.tsx [hunk 4]
```diff
@@ -1192,6 +1199,40 @@ const ResourceAllocationFormItems: React.FC<
...
       {/* TODO: Support cluster mode */}
+      {!baiClient._config.hideAgents && (
+        <Form.Item
+          label={t('session.launcher.SelectAgent')}
+          required
+          tooltip={<Trans i18nKey={'session.launcher.DescSelectAgent'} />}
+        >
+          <Flex gap={'xs'}>
+            <Form.Item required noStyle style={{ flex: 1 }} name="agent">
+              <AgentSelector
```
Comment (agatha197) [inline react/src/components/ResourceAllocationFormItems.tsx]:
```suggestion
              <AgentSelect
```

File: react/src/components/ResourceAllocationFormItems.tsx [hunk 5]
```diff
@@ -1192,6 +1199,40 @@ const ResourceAllocationFormItems: React.FC<
...
+                onChange={(value, option) => {
+                  if (value !== 'auto') {
+                    form.setFieldsValue({
+                      cluster_mode: 'single-node',
+                      cluster_size: 1,
+                    });
+                  }
+                  // TODO: set cluster mode to single node and cluster size to 1 when agent value is not "auto"
+                }}
+              ></AgentSelector>
```
Comment (agatha197) [inline react/src/components/ResourceAllocationFormItems.tsx]:
```suggestion
              />
```

**Review by agatha197 — CHANGES_REQUESTED** [review]
AgentSelector still remains.

**Review by agatha197 — APPROVED** [review]
Let's handle the resource preset issue separately. LGTM

**Pull request merged**